### PR TITLE
PAYARA-3450 Integrate Jakarta 4.02 version of Servlet

### PR DESCRIPTION
**Pull request by arjantijms**

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Jenkins test

File: appserver/security/jacc.provider.file/pom.xml [hunk 1]
```diff
@@ -80,30 +78,32 @@
...
-            <groupId>javax.security.jacc</groupId>
-            <artifactId>javax.security.jacc-api</artifactId>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
         </dependency>
         <dependency>
-             <groupId>javax.servlet</groupId>
-             <artifactId>javax.servlet-api</artifactId>
+            <groupId>javax.security.jacc</groupId>
+            <artifactId>javax.security.jacc-api</artifactId>
```
Comment (Cousjava) [inline appserver/security/jacc.provider.file/pom.xml]: Is there a reason for rearranging the order of dependencies?

File: appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java [hunk 2]
```diff
@@ -54,63 +54,63 @@
...
-            //System.out.println("helloAsync() says : " + result);
-            test_result = true;
-	    } catch(ExecutionException e) {
-            test_result = false;
-            System.out.println("Got async ExecutionException. Cause is " +
-                e.getCause().getMessage());
-                e.getCause().printStackTrace();
-	    }
-        Assert.assertEquals(test_result, true,"Unexpected Results");
+//        boolean test_result = false;
```
Comment (Cousjava) [inline appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java]: Don't comment out tests

[the remark this answers is not in the document]
Comment (Pandrex247) [reply appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java]: ninja'd

[the remark this answers is not in the document]
Comment (smillidge) [reply appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java]: Yes this change is unrelated to the substantive change in the PR. Also why has this test started to become unstable it's been around a long time.

[the remark this answers is not in the document]
Comment (Pandrex247) [reply appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java]: It's not immediately apparent
Looking at the Jenkins build history it started popping up after either the changes made [here ](https://jenkins.payara.fish/job/PayaraNightlyBuilder/98/) or [here](https://jenkins.payara.fish/job/PayaraNightlyBuilder/99/), so it's likely caused by one of those (hard to tell though due to the intermittent nature of the failure)

File: appserver/security/jacc.provider.file/pom.xml (hunk 1 above)
Comment (arjantijms) [reply appserver/security/jacc.provider.file/pom.xml]: Yes absolutely. The jakarta servlet dependency should take precedence. A couple of other dependencies still refer to the older javax dependency, and in their current order they would have precedence.

**Comment by arjantijms** [comment]
Jenkins test

**Comment by arjantijms** [comment]
Parallel PR in GlassFish 5.1:

https://github.com/eclipse-ee4j/glassfish/commit/cb5bc8e1c1b8d5766b18cdfccc3facc9224ca0e1

This one uses exclusions instead of ordering, and changed the api package here:

https://github.com/eclipse-ee4j/glassfish/commit/cb5bc8e1c1b8d5766b18cdfccc3facc9224ca0e1#diff-a93fde4983f83219e364fad801d455a3R318

File: appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarkWareham) [reply appserver/tests/quicklook/ejb/remoteview/src/test/RemoteViewTestNG.java]: QA-171 is tracker for test.

**Comment by arjantijms** [comment]
I'll close the PR for now. Maybe will redo it later.

**Pull request closed without merging**